extensions/sql-database-projects/src/models/project.ts

```diff
@@ -316,7 +316,8 @@ export class Project implements ISqlProject {
 		const normalizedRelativeFolderPath = utils.convertSlashesForSqlProj(relativeFolderPath);
 
 		// check if folder already has been added to sqlproj
-		const existingEntry = this.files.find(f => f.relativePath.toUpperCase() === normalizedRelativeFolderPath.toUpperCase());
+		const expectedExistingEntryRelativePath = utils.ensureTrailingSlash(normalizedRelativeFolderPath.toUpperCase());
+		const existingEntry = this.files.find(f => utils.ensureTrailingSlash(f.relativePath.toUpperCase()) === expectedExistingEntryRelativePath);
```

Comment:
I was thinking about "normalizing" (adding trailing slash) the actual folder that is being added, but decided against it as there were previous concerns about changing user inputs. With that, we can have folders like `a/b` and `a/b/` and they should match, so I have to ensure trailing slash for every record in the project during comparison.